Sources/OpenFeature/OpenFeatureClient.swift

```diff
@@ -68,58 +68,83 @@ extension OpenFeatureClient {
         defaultValue: T,
         options: FlagEvaluationOptions?
     ) -> FlagEvaluationDetails<T> {
-        let options = options ?? FlagEvaluationOptions(hooks: [], hookHints: [:])
-        let hints = options.hookHints
-        let context = openFeatureApi.getEvaluationContext()
+        let openFeatureApiState = openFeatureApi.getState()
+        var details = FlagEvaluationDetails(flagKey: key, value: defaultValue)
+        switch openFeatureApiState.providerStatus {
+        case .fatal:
+            details.errorCode = .providerFatal
+            details.errorMessage = OpenFeatureError
+                .providerFatalError(message: "unknown")
+                .description  // TODO Improve this message with error details
```

Comment:
Is this TODO something you'd like to address in this PR? If not, could you please create a follow-up issue?